cognite/client/utils/_time.py

```diff
@@ -542,23 +551,31 @@ def pandas_date_range_tz(start: datetime, end: datetime, freq: str, inclusive: s
 
 
 def validate_timezone(start: datetime, end: datetime) -> ZoneInfo:
-    ZoneInfo = import_zoneinfo()
-    pd = cast(Any, local_import("pandas"))
-
-    if missing := [name for name, timestamp in zip(("start", "end"), (start, end)) if not timestamp.tzinfo]:
+    if (start_tz := start.tzinfo) is None or (end_tz := end.tzinfo) is None:
+        missing = [name for name, timestamp in zip(("start", "end"), (start, end)) if not timestamp.tzinfo]
         names = " and ".join(missing)
         end_sentence = " do not have timezones." if len(missing) >= 2 else " does not have a timezone."
         raise ValueError(f"All times must be time zone aware, {names}{end_sentence}")
 
-    is_start_valid = isinstance(start, pd.Timestamp) or isinstance(start.tzinfo, ZoneInfo)  # type: ignore [arg-type]
-    is_end_valid = isinstance(end, pd.Timestamp) or isinstance(end.tzinfo, ZoneInfo)  # type: ignore [arg-type]
-    if not is_start_valid or not is_end_valid:
-        raise ValueError("Only pandas.Timestamp or datetime with ZoneInfo implementation of tzinfo are supported.")
+    # There are several ways to pass the same timezone unfortunately (without it being a user error):
+    # Pandas uses 'pytz' under the hood except for UTC, then it uses built-in `datetime.timezone.utc`...
+    # except when given something concrete like pytz.UTC or ZoneInfo(...). Converting to ZoneInfo via
+    # string is safe, all return timezone 'key':
+    ZoneInfo = import_zoneinfo()
+    tz_matches = start_tz is end_tz
```

Comment:
Isn't this line unnecessary?

Reply:
```suggestion
```

Reply:
Could raise `UnboundLocalError` if removed. But I agree it looked funky so I decided to rewrite it and split the do-tz-compare-equal logic out 👌

Reply:
(Kept all raise-statements in the original function)